### PR TITLE
Monster chase bridge puzzle

### PR DESCRIPTION
**Pull request by 1uk45**
Chased by monster to elevator. Needs fix for the monster to stop at position in front of elevator.

### CODE REVIEW

File: Game/Resources/NewAssets/Scripts/MonsterRun.as [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+				monsterPos.x -= speed * deltaTime;
+		
+				self.SetWorldPosition(monsterPos); 
+			}
+			if(monsterPos.x < -47.0f){
+				animCtrl.SetBool("E", true);
+			}
+
+		}
+
```
Comment (ludvigarlebrink) [inline Game/Resources/NewAssets/Scripts/MonsterRun.as]: Tabs.

File: Game/Resources/Opening Scene/Scripts/AnimationScript.as [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        RegisterUpdate();
+    }
+
+    // Called by the engine for each frame.
+    void Update(float deltaTime) {
+    }
+	
+	void TriggerAnim(){
+		animCtrl.SetBool("D", true);
+	}
```
Comment (ludvigarlebrink) [inline Game/Resources/Opening Scene/Scripts/AnimationScript.as]: Tabs.

File: Game/Resources/Constant Scene/Minecart/Script/Camera.as [hunk 3]
```diff
@@ -25,9 +25,9 @@ class Camera{
     void Update(float deltaTime) {
         // Only control camera with mouse if we're not running in VR.
         if (!IsVRActive()){
-            //MouseUpdate();
-            self.RotateYaw(0.02f);
-            self.position.y = -2;
+            MouseUpdate();
```
Comment (NinjaDanz3r) [inline Game/Resources/Constant Scene/Minecart/Script/Camera.as]: Revert this to automatic rotation.

File: src/Engine/Component/AnimationController.cpp [hunk 4]
```diff
@@ -54,41 +54,56 @@ void Component::AnimationController::UpdateAnimation(float deltaTime) {
...
                 }
             }
         }
     }
 
+	float time = 0.1f;
+	if (!activeAction1->isPlaybackModifierStatic)
+		time = deltaTime * 24.f * controller->floatMap[activeAction1->playbackModifierFloatIndex]->value;
+	else
+		time = deltaTime * 24.f * activeAction1->playbackModifier;
```
Comment (ludvigarlebrink) [inline src/Engine/Component/AnimationController.cpp]: Tabs.

File: src/Engine/Component/AnimationController.cpp [hunk 5]
```diff
@@ -54,41 +54,56 @@ void Component::AnimationController::UpdateAnimation(float deltaTime) {
...
-                    activeTransition->transitionProcess = 0.f;
-                    break;
                 }
             }
         }
     }
 
+	float time = 0.1f;
+	if (!activeAction1->isPlaybackModifierStatic)
+		time = deltaTime * 24.f * controller->floatMap[activeAction1->playbackModifierFloatIndex]->value;
```
Comment (ludvigarlebrink) [inline src/Engine/Component/AnimationController.cpp]: Tabs.

File: src/Editor/GUI/Editors/AnimationControllerEditor.cpp [hunk 6]
```diff
@@ -113,8 +113,8 @@ void AnimationControllerEditor::ShowNode(Node* node) {
 
             if (ImGui::BeginPopup("Select float##Float")) {
                 ImGui::Text("Select float");
-
-                for (std::size_t i = 0; i < animationController->boolMap.size(); ++i)
+				ImGui::Separator();
```
Comment (ludvigarlebrink) [inline src/Editor/GUI/Editors/AnimationControllerEditor.cpp]: Tabs.

File: src/Editor/GUI/Editors/AnimationControllerEditor.cpp [hunk 7]
```diff
@@ -113,8 +113,8 @@ void AnimationControllerEditor::ShowNode(Node* node) {
...
+                for (std::size_t i = 0; i < animationController->floatMap.size(); ++i)
```
Comment (ludvigarlebrink) [inline src/Editor/GUI/Editors/AnimationControllerEditor.cpp]: Tabs.

File: src/Editor/GUI/Editors/AnimationControllerEditor.cpp [hunk 8]
```diff
@@ -167,7 +167,7 @@ void AnimationControllerEditor::ShowNode(Node* node) {
 
             if (ImGui::BeginPopup("Select bool##Bool")) {
                 ImGui::Text("Select bool");
-
+				ImGui::Separator();
```
Comment (ludvigarlebrink) [inline src/Editor/GUI/Editors/AnimationControllerEditor.cpp]: Tabs,

File: Game/Resources/NewAssets/Scripts/Bridge.as [hunk 9]
```diff
@@ -1,12 +1,14 @@
 class Bridge {
     Entity @self;
     Entity @cart;
+	Entity @monster;
```
Comment (Ittis) [inline Game/Resources/NewAssets/Scripts/Bridge.as]: incorrect indentation

File: Game/Resources/NewAssets/Scripts/Bridge.as [hunk 10]
```diff
@@ -1,12 +1,14 @@
...
     bool trigger;
     bool playingAudio;
 
     Bridge(Entity @entity){
         @self = @entity;
         @cart = GetEntityByGUID(1510920517);
+		@monster = GetEntityByGUID(1512394369);
```
Comment (Ittis) [inline Game/Resources/NewAssets/Scripts/Bridge.as]: incorrect indentation

File: Game/Resources/NewAssets/Scripts/Bridge.as [hunk 11]
```diff
@@ -25,7 +27,10 @@ class Bridge {
         else if (roll(self.rotation) > 0.0f) {
             trigger = false;
             SendMessage(cart, 3);
+			monster.SetEnabled(true, true);
+			SendMessage(monster, 0);
```
Comment (Ittis) [inline Game/Resources/NewAssets/Scripts/Bridge.as]: incorrect indentation

File: Game/Resources/NewAssets/Scripts/Bridge.as [hunk 12]
```diff
@@ -25,7 +27,10 @@ class Bridge {
...
         }
+		SendMessage(monster, 0);
```
Comment (Ittis) [inline Game/Resources/NewAssets/Scripts/Bridge.as]: incorrect indentation

File: Game/Resources/NewAssets/Scripts/MonsterRun.as [hunk 13]
```diff
@@ -0,0 +1,52 @@
+class MonsterRun {
+    Hub @hub;
+    Entity @self;
+	vec3 monsterPos;
+	float speed;
```
Comment (Ittis) [inline Game/Resources/NewAssets/Scripts/MonsterRun.as]: incorrect indentation

File: Game/Resources/NewAssets/Scripts/MonsterRun.as [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+	bool run;
+	Component::AnimationController @animCtrl;
+	
+    MonsterRun(Entity @entity){
+        @hub = Managers();
+        @self = @entity;
+		speed = 5.0f;
+		run = false;
+		@animCtrl = self.GetAnimationController();
```
Comment (Ittis) [inline Game/Resources/NewAssets/Scripts/MonsterRun.as]: incorrect indentation

File: src/Engine/Component/AnimationController.cpp (hunk 4 above)
Comment (ludvigarlebrink) [reply src/Engine/Component/AnimationController.cpp]: This seems outdated.

**Pull request merged**